.github/pull_request_template.md

```diff
@@ -0,0 +1,36 @@
+## Description
+
+> Provide a brief description about **why** this PR is necessary. Be sure to provide context.
+
+## Related issues and/or PRs
+
+> If this PR addresses or references any issues and/or other PRs, list them here. For example, "Fixes #123", "Related to #456", or "Addresses #789".
+
+## Changes made
+
+> Outline the specific changes made in this pull request. Include relevant details, such as added features, bug fixes, code refactoring, or improvements.
+
+## How these changes were tested
+
+> Describe the testing that you have performed to ensure the changes are working as expected. Include details about the test cases or scenarios covered.
+
+## Checklist
```

Comment:
This checklist is basically just heads-up (and best-effort), or it shouldn't be violated? For instance, we might need to work on some PRs in parallel and we couldn't meet `Any dependent changes in other PRs have been merged and published` in this case. I think we can go without checking the item by writing something about it in `Additional notes`. What do you think?

Another question is whether we should check checkboxes which are not related to a PR (e.g., `Tests have been added for the changes` should be checked for document updates, or not 🤔 ) ?

Reply:
@komamitsu Thanks for the questions!

> This checklist is basically just heads-up (and best-effort), or it shouldn't be violated? For instance, we might need to work on some PRs in parallel and we couldn't meet `Any dependent changes in other PRs have been merged and published` in this case. 

In general, the checklist is mostly a heads up since there's nothing preventing the PR from being merged if the boxes aren't checked. 

> I think we can go without checking the checkbox by writing something about it in `Additional notes`. What do you think?

I think we can note that kind of guideline in our team handbook.

> Another question is whether we should check checkboxes which are not related to a PR (e.g., `Tests have been added for the changes` should be checked for document updates, or not 🤔 ) ?

I think this is another situation where we can mention in the `Additional notes` section something like `Unchecked boxes in the checklist do not apply to this PR.` I'll note this in the team handbook as well.

Reply:
How about noting these on this template so that we don't need to look at the team handbook? Also, ScalarDB is OSS and those who can't access the handbook and want to contribute would be able to know the guideline if it's written on this template.

Reply:
@komamitsu I'm OK with adding it to this PR template. Do you feel that the following description addresses both of your concerns?:

```suggestion
## Checklist

> This checklist is provided on a best-effort basis. If any items in this list are not applicable to this PR or are dependent on other unmerged PRs, please don't mark the checkboxes. Instead, leave note about why the items are not applicable in the "Additional notes" section.
```

If this looks OK, I'll add this description to the other PRs that I have submitted for this PR template.

Reply:
@josh-wong Looks perfect 💯, thank you!

Reply:
@josh-wong Sorry for getting back here. On second thought, it's often that we update only document with no need to write tests, check warnings and so on. How about checking checkboxes without writing down something in `Additional notes` if it's N/A or not related to the PR?

Reply:
@komamitsu No worries! I'm OK with that too since I don't think the impact would be big either way (but this way may reduce time spent on writing the PR). 

How about something like the following?:

```suggestion
## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.
```

Reply:
💯  Thank you!

Reply:
@josh-wong Can you apply the changes to the other repos?
If it's reviewed and merged, then the other related PRs can be merged without review.